icu4c/source/common/dictbe.cpp

```diff
@@ -1090,6 +1097,9 @@ static inline bool isKatakana(UChar32 value) {
             (value >= 0xFF66 && value <= 0xFF9f);
 }
 
+static inline bool isHiragana(UChar32 value) {
+    return value >= 0x3041 && value <= 0x309F;
```

Comment:
use UnicodeSet instead. See https://github.com/unicode-org/icu/commit/704415402a68be8bbc143e5c78c3102b0d40ba90
about the right way to set up Unicode set. If you need to check isHiragana. then you need to
1.  add   UnicodeSet                fHiraganaWordSet; back to CjkBreakEngine in the .h file
2.  in the constructor
fHiraganaWordSet.applyPattern(UnicodeString(u"[:Hiragana:]"), status);
fHiraganaWordSet.compact();
3. in the code where you need to check that char is a hiragana
instead of isHiragana(c), call fHiraganaWordSet.contains(c)

See https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/classicu_1_1UnicodeSet.html

Reply:
Resolved